aptos-move/move-examples/staking/sources/commission.move

```diff
@@ -0,0 +1,119 @@
+        // Timestamp for tracking yearly commission.
+        last_update_secs: u64,
+        // Amount of debt in dollars owed to the operator due to insufficient amount received from node commission.
+        // This can happen if the commission rate is set too high or APT price is too low.
+        commission_debt: u64,
+    }
+
+    fun init_module(commission_signer: &signer) {
+        let signer_cap = resource_account::retrieve_resource_account_cap(commission_signer, @manager);
+        move_to(commission_signer, CommissionConfig {
+            manager: @manager,
+            operator: @operator,
+            yearly_commission_amount: INITIAL_COMMISSION_AMOUNT,
+            signer_cap,
+            last_update_secs: timestamp::now_seconds(),
+            commission_debt: 0,
+        });
+    }
+
+    #[view]
+    public fun yearly_commission_amount(): u64 acquires CommissionConfig {
+        (&CommissionConfig[@staking]).yearly_commission_amount
+    }
+
+    #[view]
+    public fun commission_owed(): u64 acquires CommissionConfig {
+        let config = &CommissionConfig[@staking];
+        // Commission earned so far = per second commission rate * seconds passed.
+        let now_secs = timestamp::now_seconds();
+        let seconds_passed = now_secs - config.last_update_secs;
+        let commission_earned = math64::mul_div(seconds_passed, config.yearly_commission_amount, ONE_YEAR_IN_SECONDS);
+
+        commission_earned + config.commission_debt
+    }
+
+    /// Can be called by the manager to change the yearly commission amount.
+    public entry fun set_yearly_commission_amount(manager: &signer, new_commission: u64) acquires CommissionConfig {
+        let config = &mut CommissionConfig[@staking];
+        assert!(signer::address_of(manager) == config.manager, EUNAUTHORIZED);
+        config.yearly_commission_amount = new_commission;
+    }
+
+    /// Can be called by the manager or operator to change the account that receives the commission.
+    public entry fun set_operator(account: &signer, new_operator: address) acquires CommissionConfig {
+        let config = &mut CommissionConfig[@staking];
+        let account_addr = signer::address_of(account);
+        assert!(account_addr == config.manager || account_addr == config.operator, EUNAUTHORIZED);
+        config.operator = new_operator;
+    }
+
+    /// Distribute the commission to operator and remaining amount to manager.
+    /// Can only be called by the manager or operator.
+    public entry fun distribute_commission(account: &signer) acquires CommissionConfig {
+        // Commission owed so far plus any debt.
+        let commission_owed = (commission_owed() as u128);
+
+        // Only manager or operator can call this function.
+        let config = &mut CommissionConfig[@staking];
+        let account_addr = signer::address_of(account);
+        assert!(account_addr == config.manager || account_addr == config.operator, EUNAUTHORIZED);
```

Comment:
Ideally put this in an inline function, as it's repeated logic around authorization, to prevent missing authorization.